src/libraries/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs

```diff
@@ -706,6 +707,59 @@ public void TestMultipleServerBind()
             connection.Timeout = new TimeSpan(0, 3, 0);
         }
 
+        [ConditionalFact(nameof(LdapConfigurationExists))]
+        public async Task TestAsyncSearchWithThreadRestart()
+        {
+            using LdapConnection connection = GetConnection();
+
+            string ouName = "ProtocolsGroup9";
+            string dn = "ou=" + ouName;
+
+            try
+            {
+                DeleteEntry(connection, dn);
+
+                AddOrganizationalUnit(connection, dn);
+                SearchResultEntry sre = SearchOrganizationalUnit(connection, LdapConfiguration.Configuration.SearchDn, ouName);
+                Assert.NotNull(sre);
+
+                string filter = "(objectClass=organizationalUnit)";
+                SearchRequest searchRequest = new SearchRequest(
+                                                        dn + "," + LdapConfiguration.Configuration.SearchDn,
+                                                        filter,
+                                                        SearchScope.OneLevel,
+                                                        null);
+
+                ASyncOperationState state = new ASyncOperationState(connection);
+                IAsyncResult asyncResult = connection.BeginSendRequest(
+                                                searchRequest,
+                                                PartialResultProcessing.ReturnPartialResultsAndNotifyCallback,
+                                                RunAsyncSearch,
+                                                state);
+
+                asyncResult.AsyncWaitHandle.WaitOne();
+                Assert.True(state.Exception == null, state.Exception == null ? "" : state.Exception.ToString());
```

Comment:
Why need to check `state.Exception == null` again in the message?

Reply:
To be honest, I'm not sure. I just replicated the pattern that was already being used for other assertions in this test class. For example: https://github.com/dotnet/runtime/blob/0afabfb44653e922b4fd9950f07e482b65ad08cd/src/libraries/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs#L457